src/main/java/org/mule/service/soap/client/SoapCxfClient.java

```diff
@@ -127,37 +130,40 @@
   }
 
   @Override
-  public void stop() throws MuleException {
-    disposeIfNeeded(dispatcher, LOGGER);
-    client.destroy();
-  }
-
-  @Override
-  public void start() throws MuleException {
-    initialiseIfNeeded(dispatcher);
+  public SoapResponse consume(SoapRequest request) {
+    return consume(request, defaultDispatcher);
   }
 
-  /**
-   * Consumes an operation from a SOAP Web Service.
-   */
   @Override
-  public SoapResponse consume(SoapRequest request) throws SoapFaultException {
+  public SoapResponse consume(SoapRequest request, MessageDispatcher dispatcher) {
+    requireNonNull(dispatcher, "Message Dispatcher cannot be null");
     String operation = request.getOperation();
     Exchange exchange = new ExchangeImpl();
-    Object[] response = invoke(request, exchange);
+    Object[] response = invoke(request, exchange, dispatcher);
     return responseGenerator.generate(operation, response, exchange);
   }
 
+  @Override
+  public void stop() throws MuleException {
+    disposeIfNeeded(defaultDispatcher, LOGGER);
+    client.destroy();
+  }
+
+  @Override
+  public void start() throws MuleException {
+    initialiseIfNeeded(defaultDispatcher);
```

Comment:
initialise during start. Is this really necessary? implement initialisable if it is.